### ISSUE TITLE
Support Multiple Nuget Feeds

### ISSUE THREAD

**Issue by BlythMeister** [CycloneDX/cyclonedx-dotnet#583]
You can set an alternative feed, but this only supports changing the feed.

It would be nice to be able to specify multiple nuget feeds.

Or, even better use the nuget.config file that projects use.

**Comment by timshepherd-academy** [comment]
I'd like to vote this one up if possible

I have a project that has packages from a internal private (Teamcity based) feed, and "public packages" from Nuget.org. The Cyclone tool keeps trying to get the Nuget packages from the internal feed - which of course fails, but then the whole process fails and the tool stops.

Publishing the internal packages to Nuget.org is not an option for us.

**Comment by Bertk** [comment]
We use **CycloneDX tool** with multiple NuGet feeds (also with private Azure DevOps feeds) without problems since years.
We use **nuget.config** file for feed configuration. 

Please make sure to execute a successful `dotnet build` before running the **CycloneDX tool**.

**Comment by timshepherd-academy** [comment]
@Bertk I'm curious to know the details of how you make that work 'cos I couldnt make the cyclone tool work as a build step in Teamcity or the commandline

**Comment by BlythMeister** [comment]
I think, if you have already done the build/restore steps using a standard .net cli and the obj project assets have already been created, then it works.

If your trying to use cyclone without 1st building (therefore meaning cyclone does the restore) it doesn't work.

**Comment by Bertk** [comment]
@timshepherd-academy  Please also use **dotnet-CycloneDX** parameter  `-dpr|--disable-package-restore`.

see also [GenerateSBOM.yml](https://github.com/Bertk/arcade-light-sample/blob/main/eng/azuredevops/pipelines/steps/GenerateSBOM.yml#L41)

**Comment by jwfx** [comment]
CycloneDX only seems to look for packages in the global package cache. But even after a full restore, certain packages will never be part of that cache.

One issue we ran into are [fallbackpackagefolders](https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#fallbackpackagefolders-section).

Only after commenting out the fallbackpackagefolders and repopulating the global cache, the CycloneDX exceptions went away.

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="DevExpress 21.2 Local" value="C:\Program Files (x86)\DevExpress 21.2\Components\System\Components\Packages" />
  </packageSources>
  <fallbackPackageFolders>
    <!--<add key="DevExpress 21.2 Local Offline Packages" value="C:\Program Files (x86)\DevExpress 21.2\Components\Offline Packages"/>-->
  </fallbackPackageFolders>
</configuration>
```

CycloneDX should not only honor the full restore hierarchy, e.g. via nuget.config, but also handle fallbackPackageFolders without throwing exceptions about missing packages.

**Comment by FredAtIvory** [comment]
I would like to see CycloneDX support multiple Nuget feeds. My project uses packages from two sources, the standard Nuget source and a private DevExpress source. When I build my project in VS 2022, I can configure multiple feeds and the package restore works fine. I can also configure multiple feeds in **dotnet restore**. But I cannot in **dotnet cyclonedx**.

My current solution is to do a **dotnet restore** (with multiple feeds) first to refresh the global package cache, and then do a **dotnet cyclonedx -dpr**. This works, but I think it would be better if you could configure multiple Nuget sources in cyclonedx.
